lib/internal/crypto/keys.js

```diff
@@ -359,13 +369,41 @@ function prepareAsymmetricKey(key, ctx) {
     // Expect PEM by default, mostly for backward compatibility.
     return { format: kKeyFormatPEM, data: getArrayBufferOrView(key, 'key') };
   } else if (typeof key === 'object') {
-    const { key: data, encoding } = key;
+    const { key: data, encoding, asymmetricKeyType: typeStr } = key;
+
+    let asymmetricKeyType;
+    if (typeStr == null)
+      asymmetricKeyType = undefined;
+    else if (typeStr === 'dh')
+      asymmetricKeyType = EVP_PKEY_DH;
+    else if (typeStr === 'dsa')
+      asymmetricKeyType = EVP_PKEY_DSA;
+    else if (typeStr === 'ec')
+      asymmetricKeyType = EVP_PKEY_EC;
+    else if (typeStr === 'ed25519')
+      asymmetricKeyType = EVP_PKEY_ED25519;
+    else if (typeStr === 'ed448')
+      asymmetricKeyType = EVP_PKEY_ED448;
+    else if (typeStr === 'rsa')
+      asymmetricKeyType = EVP_PKEY_RSA;
+    else if (typeStr === 'rsa-pss')
+      asymmetricKeyType = EVP_PKEY_RSA_PSS;
+    else if (typeStr === 'sm2')
+      asymmetricKeyType = EVP_PKEY_SM2;
+    else if (typeStr === 'x25519')
+      asymmetricKeyType = EVP_PKEY_X25519;
+    else if (typeStr === 'x448')
+      asymmetricKeyType = EVP_PKEY_X448;
+    else
+      throw new ERR_INVALID_ARG_VALUE('options.asymmetricKeyType', typeStr);
```

Comment:
A switch statement would be nicer here from a code readability point of view.

Reply:
I suggest key-map object instead:
```js
const keyTypeMap = {
  'dh': EVP_PKEY_DH,
  'dsa': EVP_PKEY_DSA,
  'ec': EVP_PKEY_EC,
  'ed25519': EVP_PKEY_ED25519,
  'ed448': EVP_PKEY_ED448,
  'rsa': EVP_PKEY_RSA,
  'rsa-pss': EVP_PKEY_RSA_PSS,
  'sm2': EVP_PKEY_SM2,
  'x25519': EVP_PKEY_X25519,
  'x448': EVP_PKEY_X448,
  null: undefined // Map null to undefined
};
let asymmetricKeyType = keyTypeMap[typeStr];
if (asymmetricKeyType === undefined) {
  throw new ERR_INVALID_ARG_VALUE('options.asymmetricKeyType', typeStr);
}
```